libdnf/rpm/transaction.cpp

```diff
@@ -920,15 +924,15 @@ void Transaction::set_script_out_fd(int fd) {
     }
     auto * script_fd = fdDup(fd);
     if (!script_fd) {
-        throw Exception("fdDup()");
+        throw TransactionError(M_("set_script_out_fd: Cannot duplicate file descriptor: {}"), Fstrerror(script_fd));
```

Comment:
Also don't use function names in user-facing error messages:
> Failed to set scriptlet output file, cannot duplicate file descriptor: {}

(or similar, if I got the details of what the code is doing wrong)

Same below on line 935.

Reply:
Ok